### PR TITLE
[FLINK-30685][runtime] Support mark the transformations whose parallelism is infected by the input transformation

### PR DESCRIPTION
**Pull request by JunRuiLee**
## What is the purpose of the change

In order to chain operators together as much as possible, many downstream operators will use the parallelism of upstream input operators in the table planner.

If some operators need to have their own defined parallelism, the parallelism will be explicitly set. Therefore, the operator that takes the parallelism of the upstream operator as its own parallelism should be automatically derived by the AdaptiveBatchScheduler.

## Brief change log

Set the parallelismConfigured of the operator whose parallelism is obtained from the parallelism of the upstream operator to false

## Verifying this change

This change is already covered by existing tests, such as e2e tpc-ds. 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not documented

### CODE REVIEW

**Comment by JunRuiLee** [comment]
@zhuzhurk Could you help to review this pr? Thanks a lot!

**Review by zhuzhurk — COMMENTED** [review]
Thanks for creating this PR! @JunRuiLee 
The change looks good to me except for 1 comment.

File: flink-end-to-end-tests/test-scripts/test_tpcds.sh
Comment (zhuzhurk) [reply flink-end-to-end-tests/test-scripts/test_tpcds.sh:66]: Sorry I missed that it was moved above.

**Review by zhuzhurk — APPROVED** [review]
LGTM. We can merge this once all the predecessor PRs are merged.

**Comment by JunRuiLee** [comment]
@flinkbot run azure

**Comment by zhuzhurk** [comment]
@flinkbot run azure

**Pull request closed without merging**
